kinto/core/resource/viewset.py

```diff
@@ -118,14 +128,17 @@ def get_view_arguments(self, endpoint_type, resource_cls, method):
         endpoint_args = getattr(self, by_method, {})
         args.update(**endpoint_args)
 
+        request_schema = args.get('schema', RequestSchema)
+
+        # Instantiate request schema
+        request_schema = request_schema()
+
         if method.lower() in map(str.lower, self.validate_schema_for):
```

Comment:
With the current implementation it still is, the `body` field is still added just on the viewset. I've tried to preserve this for backward compatibility, but we can indeed solve this on the schemas.